### DISCUSSION TITLE
Accessing a number using the array operator e.g. 1[0] returns undefined

### DISCUSSION THREAD

**Discussion by cmdcolin** [denysdovhan/wtfjs#325] [WTF Questions]
I stumbled on this one today. we had some type confusion and i was accessing a number with a array operator. I am not sure why this returns undefined though! some sort of coercion occurs maybe? but then why do we get undefined at the end?

**Comment by ericblade** [comment]
Numbers are objects of class Number.  Objects can have properties, and you can access them with []
... something like that :-)

**Reply by cmdcolin** [reply]
would be interested in verification of that somehow, my cursory googling is that number is truly a primitive so it would have to be coerced to a object somehow in this process

**Reply by ericblade** [reply]
1["x"] also equals undefined.  Pretty much everything can have properties, and so accessing a property with [] can be a thing on pretty much everything.

I don't have a specific explanation for what happens in the middle :D

looks like the setter for properties on numbers doesn't actually set anything, though
![image](https://github.com/denysdovhan/wtfjs/assets/1451847/e89816b1-2714-45f4-a52a-1fdf08920218)
